src/with-urql-client.tsx

```diff
@@ -76,12 +81,18 @@ function withUrqlClient<T = any, IP = any>(
     const displayName = Page.displayName || Page.name || 'Component';
     withUrql.displayName = `withUrqlClient(${displayName})`;
 
-    withUrql.getInitialProps = async (ctx: NextPageContext) => {
+    withUrql.getInitialProps = async (ctx: NextPageContext | AppContext) => {
       const { AppTree } = ctx;
 
+      const appCtx = (ctx as AppContext).ctx;
+      const isApp = !!appCtx;
```

Comment:
This handles checking for the proper `ctx` object in the case where we're wrapping `_app.js` (`ctx,ctx`) or just a page component (`ctx`).